src/dataflow/src/source/file.rs

```diff
@@ -257,37 +299,40 @@ pub fn read_file_task<Ctor, I, Out, Err>(
 
             #[cfg(target_os = "linux")]
             {
-                let mut inotify = match Inotify::init() {
-                    Ok(w) => w,
-                    Err(err) => {
-                        error!("file source: failed to initialize inotify: {:#}", err,);
-                        return;
-                    }
-                };
-                if let Err(err) = inotify.add_watch(&path, WatchMask::ALL_EVENTS) {
-                    error!(
-                        "file source: failed to add watch for file: {:#} (path: {})",
-                        err,
-                        path.display()
-                    );
-                    return;
-                }
-                let path = path.clone();
+                let mut inotify = Inotify::init()
+                    .with_context(|| format!("file source: failed to initialize inotify"))?;
+                inotify
+                    .add_watch(&_path, WatchMask::ALL_EVENTS)
+                    .with_context(|| format!("failed to add watch for file {}", _path.display()))?;
```

Comment:
Can we get the underlying OS error message in here as well:

```
inotify.td:8:1: error: executing query failed: db error: ERROR: Source error: File IO: failed to add watch for file /tmp/.tmps0gOpN/dynamic.csv
```

The previous code had ```..."file source: failed to add watch for file: {:#} (path: {})", err,...``` which provided that information.

Reply:
Hi Philip, it looks like it's a real pain to add this context back into the error message in a way that is printed by our current error handling. I believe I know how to have this error message printed, but it requires updating tests that read error messages. I'm going to file a bug to track getting this error message printed correctly.

Reply:
@cirego No problem, thanks for opening the new ticket.